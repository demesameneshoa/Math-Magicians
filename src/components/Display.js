import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ result }) => (
  <div className="Display">
    {result}
  </div>
);

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
