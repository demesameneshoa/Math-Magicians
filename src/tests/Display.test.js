import React from 'react';
import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Display from '../components/Display';

describe('Display component unit test', () => {
  test('Display Component Renders Properly ', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('The Display Components Displays Result', () => {
    render(<Display result="15.7" />);
    expect(screen.getByText('15.7')).toBeInTheDocument();
  });
});
