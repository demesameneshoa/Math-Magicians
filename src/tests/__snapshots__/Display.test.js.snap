// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component unit test Display Component Renders Properly  1`] = `
<div
  className="Display"
>
  0
</div>
`;
